javascript/javascript1/week3/notepad.js

```diff
@@ -0,0 +1,64 @@
+//Save a note
+const notes = [];
+
+function saveNote(content, id) {
+  notes.push({content :content, id: id})
+}
+
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+
+console.log(notes);
+
+//Get a note
+function getNote(id) {
+  for(let i =0; i<notes.length ;i++){
+    if(notes[i].id === id){
+      return notes[i].content;
+    }
+  }
+  return 'The id is not valid!';
+}
+
+const firstNote = getNote(2);
+console.log(firstNote);
+
+//Log out notes
+function logOutNotesFormatted() {
+  for (let i =0; i<notes.length ;i++){
+    console.log ( `The note with id: ${notes[i].id}, has the following note text: ${notes[i].content}.`);
+  }
+  
+}
+
+logOutNotesFormatted();
+
+//Unigue feature
+//Adding new key-value pair to indicate task completion.
+
+function markTaskCompleted (id){
+  for(let i =0; i<notes.length ;i++){
+    if(notes[i].id === id){
+      notes[i].completed = true;
+    }
+  }
+  return notes;
+}
+markTaskCompleted(2)
+console.log(notes)
+
+//Checking if a task is completed or not
+function checkTaskCompleted(content){
```

Comment:
Good job with your unique feature! ⭐ 

What would happen if a developer in the future set `note.completed` to `false`? 🐛 Would that behave as you expect?

In JavaScript, `undefined` is often a nuisance that wreak havoc on your code base. I recommend avoiding relying on it if you can get away with it. Perhaps `note.completed` should be set to a value when notes are added? 😄 

As a minor note, it might make more sense to check by ID instead of content, but I assume this was a deliberate choice? 👀

Reply:
Thank you⭐ I see what you mean about undefined, so I've initialized note.completed as false when adding notes.

I used content to check tasks because I thought it would be easier to remember tasks by name (like checking if I washed the dishes) 😅, but I see that using ID might be more reliable.

Reply:
I can definitely agree with that, haha. In practice, you would probably have a search function you could get the ID with or perhaps have the notes listed in a grid where the user clicks (where the ID would also be available to us then without the user having to know it). 😄